surefire-providers/surefire-junit-platform/src/main/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProvider.java

```diff
@@ -275,10 +276,10 @@ private LauncherDiscoveryRequest buildLauncherDiscoveryRequestForRerunFailures(
                 .map( TagFilter::excludeTags )
                 .ifPresent( filters::add );
 
-        TestListResolver testListResolver = parameters.getTestRequest().getTestListResolver();
-        if ( !testListResolver.isEmpty() )
+        TestListResolver filter = optionallyWildcardFilter( parameters.getTestRequest().getTestListResolver() );
+        if ( !filter.isEmpty() && !filter.isWildcard() )
```

Comment:
@slawekjaranowski 
No it is not same. See the other Providers -> JUnit4Provider https://github.com/apache/maven-surefire/blob/master/surefire-providers/surefire-junit4/src/main/java/org/apache/maven/surefire/junit4/JUnit4Provider.java#L436 and JUnitCoreProvider https://github.com/apache/maven-surefire/blob/master/surefire-providers/surefire-junit47/src/main/java/org/apache/maven/surefire/junitcore/JUnitCoreProvider.java#L274
It is there sine 2015, i.e. 7 years.

Reply:
ok,

```java

    public static TestListResolver optionallyWildcardFilter( TestListResolver resolver )
    {
        return resolver.hasMethodPatterns() ? resolver : WILDCARD;
    }

    public final boolean isWildcard()
    {
        return equals( WILDCARD );
    }
```

so:

```java
TestListResolver filter = optionallyWildcardFilter( resolver );

// this expression will always true - or I don't see something
!filter.isWildcard()  == resolver.hasMethodPatterns()

```

and is it possible?

```java
resolver.hasMethodPatterns() == true && resolver.isEmpty() == true
```

Reply:
Hi @slawekjaranowski ,

Regarding `return resolver.hasMethodPatterns() ? resolver : WILDCARD;` it won't be right because there are many patterns, and not only method patterns, there are also class patterns. So, this proposal won't work properly. Thx for this remark.
The reason why we call `optionallyWildcardFilter()` is the fact we do not need to care about filtering if it is empty or wildcard however it is a difference for the `TestListResolver` itself but no difference for provider purposes.

Reply:
We are again in the discussions with refactoring other providers. But this issue is about Junit5 provider. So my goal is not to make all these providers very different. I have made them similar in this PR through JunitPlatformProvider changes and nowhere else. I am not interested in others yet.
The `optionallyWildcardFilter()` can be removed but the condition can be changed in the next PR, i.e. `!filter.isEmpty() && !filter.isWildcard() && !filter.hasMethodPatterns()`. Here I am trying to make a fix and not to make providers too different in terms of filtering. Feel free to open a new PR after this and you can change all providers.

Reply:
@slawekjaranowski 
If you would work on the change, pls remove the method `getWildcard()` in TLR. It is not used.

Reply:
ok, does not matter how and what we change in given expression, condition - such change should impact on unit tests. 

- I suppose that some of unit tests or assertions are missing, so should be fixed to be sure what we want to achieve.

- If we sure that unit tests cover in proper way - the change has no effect and is not needed.

Unit tests exist in `JUnitPlatformProviderTest`

Reply:
@slawekjaranowski 
I guess `JUnitPlatformProviderTest` has similar use case which we can copy and adapt, this may help. This could be done in advance, your right. Please help me and push a new commit with the test if you have some spare time. Yesterday and today I am supporting Olivier with testing on a demo project. I would like to cut the release, the jdk 18 GA will be released soon as well and we have jdk 18 related fixes.

Reply:
As I understand `getTestListResolver()` returns `TestListResolver` with informations based on  `-Dtest` param.

So please give me example like:

`test=xxxx` then filters.contains or filters.notContains some of TestMethodFilter instance

then I will can write proper unit tests

Reply:
@slawekjaranowski 
I approved #491  maybe too early, we need to check the Javadoc by the compiler with `<arg>-Xdoclint:all</arg>`. Checking it during Site is too late. The contributors can check it by usual build `mvn install`.

Reply:
> As I understand getTestListResolver() returns TestListResolver with informations based on -Dtest param.

It also has some form of merged patterns includes/excludes, includeFile/excludeFile.
If you remember SUREFIRE-1426 with ildefonso and Olivier, that's exactly the part of code which updated the merge logic.

> test=xxxx then filters.contains or filters.notContains some of TestMethodFilter instance or another filter instance which should be added for given -Dtest

Yes, I think the same, it would be enough to test `newFilters()` and verify the collection content. I found some example which would help you, and you can reuse it, see the test method `onlyGroupsIsDeclared()`.